start

```diff
@@ -3,15 +3,37 @@
 
 set -e
 
-# Unless the RunDevServer binary is available, we rebuild the .envrc cache with nix-shell
-# and config cachix for using our binary cache
-command -v RunDevServer >/dev/null 2>&1 \
-    || { echo "PATH_add $(nix-shell -j auto --cores 0 --run 'printf %q $PATH')" > .envrc; }
+# We start lorri to automatically update the dev environment
+# NB: Only run lorri watch if the lorri daemon isn't active
+if ! pgrep -f "lorri daemon"
+then
+    lorri watch &
+    trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+fi
 
 # Now we have to load the PATH variable from the .envrc cache
 direnv allow
 eval "$(direnv hook bash)"
 eval "$(direnv export bash)"
 
-# Finally start the dev server
-RunDevServer
+# We start the dev server that we have in our environment
+# NB: We use which so that process includes the nix store path
+$(which RunDevServer) &
+trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+
+# Finally we monitor for changes in the dev server path
+# If lorri gives us a new version, we automatically switch to the new one
```

Comment:
I'd rather implement this using a file watcher on `default.nix` [inside the DevServer](https://github.com/digitallyinduced/ihp/blob/master/IHP/IDE/DevServer.hs). Or do you see any problems with using the DevServer for that?

The problem with the start script is that it's very hard to update as it's part of the project and not IHP itself.

Reply:
So I don't think just a file watcher is enough, because we need to wait for the evaluation to complete and the `direnv` context to change. Do you know if it's easy to have `DevServer` kill itself after calling a replacement version of itself. I've not tried that before, which is why I went with the script, but I see the reasoning to put it all in `ihp`.

Reply:
We could use the [`stop`](https://github.com/digitallyinduced/ihp/blob/master/IHP/IDE/DevServer.hs#L38) function to stop all running services of the dev server. After that we could use something like `Process.proc "./start"` to start the new process and then let the new process use the current stdout, stderr and stdin. Then we'd have two processes running actually (the old one who called `stop` and then `./start` and the new one who actually does all the work).

Reply:
Yeah, sounds reasonable - we could even call `RunDevServer` directly, so we don't have to worry about `./start` running multiple `lorri watch` instances

Reply:
Yes good point :)